File: src/main/java/com/puhovin/intellijplugin/twm/settingsmanager/GlobalToolWindowManagerService.java
package com.puhovin.intellijplugin.twm.settingsmanager;

import com.intellij.openapi.components.Service;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.puhovin.intellijplugin.twm.model.ToolWindowPreferenceStore;

@Service(Service.Level.APP)
@State(name = "toolwindow-manager-global-settings", storages = @Storage("toolwindow-manager-global-settings.xml"))
public final class GlobalToolWindowManagerService implements SettingsManager {
    private ToolWindowPreferenceStore globalState = new ToolWindowPreferenceStore();

}

File: src/main/kotlin/com/puhovin/intellijplugin/twm/settingsmanager/GlobalToolWindowManagerService.kt
package com.puhovin.intellijplugin.twm.settingsmanager

import com.intellij.openapi.components.Service
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage

@Service(Service.Level.APP)
@State(name = "toolwindow-manager-global-settings", storages = [Storage("toolwindow-manager-global-settings.xml")])
class GlobalToolWindowManagerService : AbstractSettingsManager()